roles/kubernetes/master/templates/manifests/kube-apiserver.manifest.j2

```diff
@@ -117,8 +117,9 @@ spec:
     - --feature-gates={{ kube_feature_gates|join(',') }}
 {% endif %}
 {% if kube_version | version_compare('v1.9', '>=') %}
-    - --requestheader-client-ca-file={{ kube_cert_dir }}/front-proxy-ca.pem
-    - --requestheader-allowed-names=front-proxy-client
+    - --requestheader-client-ca-file={{ kube_cert_dir }}/{{ kube_front_proxy_ca }}
+{# intentionally empty #}
+    - --requestheader-allowed-names=
```

Comment:
I'm not 100% certain, but leaving `--requestheader-allowed-names` empty authorizes all CNs as users if the client cert is validated by `--requestheader-client-ca-file` which is potentially too open. kubeadm sets this to `front-proxy-client` which seems to be working now that #2352 and #2667 are resolved.

Reply:
I was running into some issues with it set the other way... I moved the front proxy to be part of the same CA as k8s in vault, so let's try adding the allowed names back